src/Service/PurchaseService.php

```diff
@@ -120,14 +120,13 @@ public function complete($data = array(), $isNotification = false)
             return $serviceResponse;
         }
 
-        // only update payment status if we're not waiting for a notification
-        if (!$serviceResponse->isAwaitingNotification()) {
+        // Only update payment status if Omnipay specifically marks it as successful
+        if ($response->isSuccessful()) {
```

Comment:
This is the critical check we’re currently missing - even if we don’t cover every other combination of redirects/notification statuses, we need to verify this before making payments as complete